lib/procedures/ensure-manta-deployment-gz-links.js

```diff
@@ -0,0 +1,185 @@
+        funcs: [
+            function theCurrServerUuid(ctx, next) {
+                sdcadm.getCurrServerUuid(function (_, currServerUuid) {
+                    ctx.currServerUuid = currServerUuid;
+                    next();
+                });
+            },
+            function getSvc(ctx, next) {
+                sdcadm.sapi.listServices({
+                    name: 'manta',
+                    application_uuid: sdcadm.sdcApp.uuid
+                }, function listSvcsCb(err, svcs) {
+                    if (err) {
+                        next(new errors.SDCClientError(err, 'sapi'));
+                    } else if (svcs.length > 1) {
+                        next(new VError('multiple "manta" services found!'));
+                    } else if (svcs.length === 0) {
+                        next(new VError('no "manta" service found on the ' +
+                            '"sdc" SAPI application'));
+                    } else {
+                        ctx.svc = svcs[0];
+                        next();
+                    }
+                });
+            },
+
+            // Find the "manta" service inst. There should be exactly one the
+            // headnode (on which we are currently running).
+            function getInst(ctx, next) {
+                sdcadm.sapi.listInstances({
+                    service_uuid: ctx.svc.uuid
+                }, function listInstCb(err, insts) {
+                    if (err) {
+                        next(new errors.SDCClientError(err, 'sapi'));
+                    } else if (!insts || insts.length === 0) {
+                        next(new VError('no "manta" instance was found'));
+                    } else if (insts.length > 1) {
+                        next(new VError(
+                            'multiple "manta" instances were found'));
+                    } else if (!insts[0].params.server_uuid) {
+                        next(new VError(
+                            '"manta" instance %s (%s) does not have ' +
+                                '"params.server_uuid" set',
+                            insts[0].uuid, insts[0].params.alias));
+                    } else if (insts[0].params.server_uuid !==
+                               ctx.currServerUuid) {
+                        next(new VError(
+                            '"manta" instance %s (%s) is not on this server: ' +
+                                'server_uuid=%s, currServerUuid=%s',
+                            insts[0].uuid,
+                            insts[0].params.alias,
+                            insts[0].params.server_uuid,
+                            ctx.currServerUuid));
+                    } else {
+                        ctx.inst = insts[0];
+                        next();
+                    }
+                });
+            },
+
+            // Create stubs in /opt/smartdc/bin/ to a few `manta-*` tools
+            // in the manta-deployment zone.
+            function createLinks(ctx, next) {
```

Comment:
Do we also need to unlink here like the sdc-headnode version does? Or are we guaranteed these are always files (not symlinks or something)?

Reply:
Ah, thanks. I'll do as you suggest.

I tested by making one of those a symlink:

```
[root@headnode (coal) /opt/smartdc/bin]# rm manta-login
[root@headnode (coal) /opt/smartdc/bin]# ln -s sdc-role manta-login
[root@headnode (coal) /opt/smartdc/bin]# ls -al manta*
-rwxr-xr-x   1 root     root         211 Nov 27 22:34 manta-adm
lrwxrwxrwx   1 root     root           8 Nov 27 22:36 manta-login -> sdc-role
-rwxr-xr-x   1 root     root         214 Nov 27 22:34 manta-oneach
```

re-running:

```
[root@headnode (coal) /opt/smartdc/bin]# sdcadm up manta --force-same-image
Finding candidate update images for the "manta" service.
Using channel dev

This update will make the following changes:
    update "manta" service to image 27b27f26-dc3c-4308-9abf-64787220c706
        (manta-deployment@master-20191126T202101Z-g764669f):
        instance "91cee159-5e36-4e2c-a310-3badb5f5e53f" (manta0) on server 564d8ce9-08d3-b66d-677f-be7573149cbe
    ensure /opt/smartdc/bin/manta-* tools are setup

Would you like to continue? [y/N] y

Create work dir: /var/sdcadm/updates/20191127T223718Z
Updating image for SAPI service "manta"
    service uuid: 54b99f0e-a568-4a5f-8bf8-64767d445b90
Installing image 27b27f26-dc3c-4308-9abf-64787220c706
    (manta-deployment@master-20191126T202101Z-g764669f)
Reprovisioning VM 91cee159-5e36-4e2c-a310-3badb5f5e53f on server 564d8ce9-08d3-b66d-677f-be7573149cbe
Waiting for manta instance 91cee159-5e36-4e2c-a310-3badb5f5e53f to come up
Wrote "/opt/smartdc/bin/manta-{login,adm,oneach}" stubs
Skipping manta-adm.1 (/opt/smartdc/man/man1/manta-adm.1 already exists)
Creating symlink "/opt/smartdc/man/man1/manta-deploy-coal.1" for "/zones/91cee159-5e36-4e2c-a310-3badb5f5e53f/root/opt/smartdc/manta-deployment/man/man1/manta-deploy-coal.1"
Skipping manta-deploy-dev.1 (/opt/smartdc/man/man1/manta-deploy-dev.1 already exists)
Skipping manta-deploy-lab.1 (/opt/smartdc/man/man1/manta-deploy-lab.1 already exists)
Skipping manta-factoryreset.1 (/opt/smartdc/man/man1/manta-factoryreset.1 already exists)
Skipping manta-init.1 (/opt/smartdc/man/man1/manta-init.1 already exists)
Skipping manta-login.1 (/opt/smartdc/man/man1/manta-login.1 already exists)
Skipping manta-oneach.1 (/opt/smartdc/man/man1/manta-oneach.1 already exists)
Updated successfully (elapsed 34s).
```

and checking:

```
[root@headnode (coal) /opt/smartdc/bin]# ls -al manta*
-rwxr-xr-x   1 root     root         211 Nov 27 22:37 manta-adm
-rwxr-xr-x   1 root     root         115 Nov 27 22:37 manta-login
-rwxr-xr-x   1 root     root         214 Nov 27 22:37 manta-oneach
```